src/app/campaign-list/campaign-list.service.ts

```diff
@@ -76,31 +97,38 @@ export class CampaignListService {
     );
   }
 
-  loadCampaignList(params?: CampaignParams) {
-    if (params) {
-      this.params = {
-        ...this.params,
-        ...(params.page && {page: params.page}),
-        ...(params.per && {per: params.per})
-      };
-    }
+  get loadedCampaigns(): Observable<Campaign[]> {
+    return this.campaigns.pipe(
+      map((campaigns: Campaign[]) => campaigns && campaigns.filter(c => !c.loading))
+    );
+  }
+
+  loadCampaignList(newParams?: CampaignParams) {
     this.count = 0;
-    this.loading = true;
-    this.flightsLoaded = 0;
 
-    // clear campaign list
-    this._campaigns.next({});
+    // newParams includes all params (from route) except per
+    this.params = {...newParams, per: this.params.per, page: (newParams && newParams.page) || 1};
+    const { page, per, advertiser, podcast, status, type, geo, zone, text, representative, before, after } = this.params;
+    const filters = this.getFilters({advertiser, podcast, status, type, geo, zone, text, representative, before, after});
 
-    this.augury.followItems(
-      'prx:campaigns',
-      {page: this.params.page, per: this.params.per}
-    ).pipe(
-      concatMap((campaignDocs: HalDoc[]) => {
-        if (campaignDocs.length) {
-          this.count = campaignDocs[0]['_count'];
-          this.total = campaignDocs[0]['_total'];
-          this.facets = campaignDocs[0]['_facets'];
-        }
+    this.augury.follow('prx:campaigns', {
+      page,
+      per,
+      ...(filters && {filters})
+    }).pipe(
+      switchMap(result => {
+        this.count = +result['count'];
+        this.total = +result['total'];
+        this.facets = result['facets'];
+        return result.followList('prx:items');
+      }),
+      switchMap((campaignDocs: HalDoc[]) => {
+        this._campaigns.next(campaignDocs.reduce((acc, doc) => {
```

Comment:
One neat trick i picked up from Peter ... 

```js
this._campaigns.next(campaignDocs.reduce((acc, doc) => ({...acc, [doc.id]: {loading: true}}), {}));
```

(That may just make me look desperate to write 1-liners).